### PR TITLE
sim: Added gcov dump on application exit.

### PR DESCRIPTION
**Pull request by fjpanag**
## Summary

Added gcov data dumping when the simulator exits.

## Impact

Gcov can be used without NSH and the gcov app.

## Testing

Tested that the simulator indeed creates the `.gcda` files on exit.
The `tools/gcov.sh` script creates a correct report.

### CODE REVIEW

**Comment by fjpanag** [comment]
I am not sure what is the style problem:

```
up_hostmisc.c:62:9: error: Mixed case identifier found
```

File: arch/sim/src/sim/win/up_hostmisc.c [hunk 1]
```diff
@@ -40,6 +48,12 @@
 
 void host_abort(int status)
 {
+#ifdef CONFIG_ARCH_COVERAGE
+  /* Dump gcov data. */
+
+  __gcov_dump();
```
Comment (xiaoxiang781216) [inline arch/sim/src/sim/win/up_hostmisc.c:54]: let's revert the change. __gcov_dump is specific to gcc/clang, so it can't work with msvc.

Comment (fjpanag) [reply arch/sim/src/sim/win/up_hostmisc.c:54]: In this case `CONFIG_ARCH_COVERAGE` shouldn't be enabled, right?

Maybe we should fix the dependencies of `CONFIG_ARCH_COVERAGE`?

Comment (xiaoxiang781216) [reply arch/sim/src/sim/win/up_hostmisc.c:54]: could be

Comment (fjpanag) [reply arch/sim/src/sim/win/up_hostmisc.c:54]: Oh, you mean specifically for Windows? And keeping only the Posix one?

Comment (xiaoxiang781216) [reply arch/sim/src/sim/win/up_hostmisc.c:54]: Yes.

**Review by xiaoxiang781216 — APPROVED** [review]
